mono/metadata/icall.c

```diff
@@ -4446,9 +4442,11 @@ ves_icall_System_Reflection_Assembly_InternalGetType (MonoReflectionAssemblyHand
 		goto fail;
 
 	/*g_print ("requested type %s in %s\n", str, assembly->assembly->aname.name);*/
-	if (!mono_reflection_parse_type (str, &info)) {
+	MonoError parse_error;
+	if (!mono_reflection_parse_type_checked (str, &info, &parse_error)) {
 		g_free (str);
 		mono_reflection_free_type_info (&info);
+		mono_error_cleanup (&parse_error);
```

Comment:
We should concat the parse error to the mono_error_set_argument below, shouldn't we?

Reply:
There's no useful message in there yet.